daphne_server/examples/service.rs

```diff
@@ -0,0 +1,82 @@
+use std::path::PathBuf;
+
+use clap::Parser;
+use daphne::DapGlobalConfig;
+use daphne_server::{router, App};
+use daphne_service_utils::Role;
+use serde::Deserialize;
+
+#[derive(Debug, Deserialize)]
+struct Config {
+    global: DapGlobalConfig,
+    port: u16,
+    role: Role,
+}
+
+impl TryFrom<Args> for Config {
+    type Error = config::ConfigError;
+    fn try_from(
+        Args {
+            configuration,
+            role,
+            port,
+        }: Args,
+    ) -> Result<Self, Self::Error> {
+        config::Config::builder()
+            .set_default("port", 3000)?
+            .add_source(match configuration {
+                Some(path) => config::File::from(path.as_ref()),
+                None => config::File::with_name("configuration"),
+            })
+            .set_override_option(
+                "role",
+                role.map(|role| config::Value::new(Some(&String::from("args.role")), role)),
+            )?
+            .set_override_option(
+                "port",
+                port.map(|port| config::Value::new(Some(&String::from("args.port")), port)),
+            )?
+            .add_source(config::Environment::with_prefix("DAP"))
+            .build()?
+            .try_deserialize()
+    }
+}
+
+#[derive(clap::Parser)]
+struct Args {
+    #[arg(short, long)]
+    configuration: Option<PathBuf>,
+    #[arg(short, long)]
+    role: Option<String>,
+    #[arg(short, long)]
+    port: Option<u16>,
+}
+
+#[tokio::main]
+async fn main() -> Result<(), Box<dyn std::error::Error + Sync + Send>> {
```

Comment:
Presumably this is meant to be copy-pastable example of a Daphne service deployment. To make this more useful, I would add a ton of comments to this function explaining what each step does. To understand the system, it should be sufficient to just go through this file.